src/lib/y2network/interfaces/bond_interface.rb

```diff
@@ -0,0 +1,51 @@
+# Copyright (c) [2019] SUSE LLC
+#
+# All Rights Reserved.
+#
+# This program is free software; you can redistribute it and/or modify it
+# under the terms of version 2 of the GNU General Public License as published
+# by the Free Software Foundation.
+#
+# This program is distributed in the hope that it will be useful, but WITHOUT
+# ANY WARRANTY; without even the implied warranty of MERCHANTABILITY or
+# FITNESS FOR A PARTICULAR PURPOSE.  See the GNU General Public License for
+# more details.
+#
+# You should have received a copy of the GNU General Public License along
+# with this program; if not, contact SUSE LLC.
+#
+# To contact SUSE LLC about this file by physical or electronic mail, you may
+# find current contact information at www.suse.com.
+
+require "y2network/interfaces/virtual_interface"
+
+module Y2Network
+  module Interfaces
+    class BondInterface < VirtualInterface
+      # Returns list of interfaces enslaved in the bond interface
+      #
+      # @return [Y2Network::InterfacesCollection] interfaces enslaved in the bond
+      # TODO: reading from backend should be handled elsewhere
+      def slaves
+        @slaves ||= bond_slaves(name).map do |iface|
+          type = Yast::NetworkInterfaces.GetType(iface)
+          Y2Network::Interface.for(iface, InterfaceType.from_short_name(type))
+        end
+
+        Y2Network::InterfacesCollection.new(@slaves)
+      end
+
+    private
+
+      # Creates list of devices enslaved in the bond device.
+      #
+      # @param bond_iface [String] a name of an interface of bond type
+      # @return list of names of interfaces enslaved in the bond_iface
+      def bond_slaves(bond_iface)
+        bond_map = Yast::NetworkInterfaces::FilterDevices("netcard").fetch("bond", {}).fetch(bond_iface, {})
+
+        bond_map.select { |k, _| k.start_with?("BONDING_SLAVE") }.values
```

Comment:
what happens if there are two bonds? It is real case in our OPS, when there are two cards to internal and two cards to external network for VPN server. or is it already selected above by that FilterDevices? But then why is needed also this select?

Reply:
selection above selects particular bond device (see last fetch)

Reply:
select is needed bcs bonding slaves are defined by options "BONDING_SLAVE0, BONDING_SLAVE1, ..." It differs from bridge where you have "BRIDGE_PORTS" with space separated list of ports / slaves.